include/rmm/mr/device/cuda_async_memory_resource.hpp

```diff
@@ -115,6 +111,26 @@ class cuda_async_memory_resource final : public device_memory_resource {
   cuda_async_memory_resource& operator=(cuda_async_memory_resource const&) = delete;
   cuda_async_memory_resource& operator=(cuda_async_memory_resource&&) = delete;
 
+  /**
+   * @brief Is cudaMallocAsync supported with this cuda runtime/driver version?
+   * @return true if both the cuda runtime and driver are newer than 11.2
+   */
+  static bool is_supported()
+  {
+    static auto runtime_version{[] {
```

Comment:
Convince me that this isn't sufficient.

```suggestion
  static bool is_supported()
  {
   #ifdef RMM_CUDA_MALLOC_ASYNC_SUPPORT
    // Check if cudaMallocAsync Memory pool supported
    auto const device = rmm::detail::current_device();
    int cuda_pool_supported{};
    auto result =
      cudaDeviceGetAttribute(&cuda_pool_supported, cudaDevAttrMemoryPoolsSupported, device.value());
    return result == cudaSuccess and cuda_pool_supported;
#else
    return false;
#endif
}
```

Reply:
You can compile with a newer cuda toolkit but run in an older runtime, then you'd get a symbol not found error.

Reply:
Which symbol would not be found? cudaDeviceAttribute will exist on all CUDA versions we support.

Reply:
`cudaDevAttrMemoryPoolsSupported`

Reply:
Once compiled that would be a value compiled into our binary that would get passed to the API, wouldn't it? Not a symbol.

Reply:
Pretty sure this code already works in the constructor.

Reply:
Hmm actually I think it's just the check doesn't work (build with cuda 11.6, run with 11.0):
```console
./gtests/CUDA_ASYNC_MR_TEST: symbol lookup error: ./gtests/CUDA_ASYNC_MR_TEST: undefined symbol: cudaMemPoolCreate, version libcudart.so.11.0
```

Reply:
I printed out the value of `cuda_pool_supported`, it's actually `1` when running under 11.0.